main.go

```diff
@@ -195,34 +195,49 @@ type BackendStats struct {
 	DowntimeStart   time.Time
 }
 
+const errMessage = `<?xml version="1.0" encoding="UTF-8"?><Error><Code>BackendDown</Code><Message>The remote server returned an error (%w)</Message><Resource>%s</Resource></Error>`
+
+func writeErrorResponse(w http.ResponseWriter, r *http.Request, err error) {
+	// Set retry-after header to indicate user-agents to retry request after 120secs.
+	// https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Retry-After
+	w.Header().Set("Retry-After", "120")
+	w.WriteHeader(http.StatusBadGateway)
```

Comment:
Is it okay to set a status code after w.Header().Set() ? let's always send a status code first

Reply:
yeah because until you do w.Write() nothing is written to the client response body @vadmeste 

You can't do a header set after the w.Write() that's all.